File: src/DesignPattern/Sigletons/SealedSingleton.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPatternCSharp.Sigletons
{
    //率先實體化
    public sealed class SealedSingleton
    {
        private static readonly SealedSingleton instance = new SealedSingleton();
        private SealedSingleton() { }
        public static SealedSingleton GetInstance() => SealedSingleton.instance;
    }
}

File: src/DesignPattern.VBNet/Sigletons/SealedSingleton.vb

Namespace DesignPattern.VBNet.Sigletons
    '率先實體化 VB.Net
    Public NotInheritable Class SealedSingleton
        Private Shared instance As SealedSingleton = New SealedSingleton()
        Private Sub New()
        End Sub
        Public Shared Function GetInstance() As SealedSingleton
            Return instance
        End Function
    End Class
End Namespace


